apps/supernova/docker/entrypoint.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+# Define the JSON file path
+APP_PROPS_FILE="/usr/share/nginx/html/appProps.json"
+
+# Copy the appProps.json file to the /usr/share/nginx/html directory if it exists
+if [ -f /appProps.json ]; then
+  cp /appProps.json "${APP_PROPS_FILE}"
+fi
+
+# Function to update JSON fields using jq
+update_json_field() {
+  field="$1"
+  value="$2"
+  if [ -n "$value" ]; then
+    # Ensure jq and temporary file handling are POSIX compliant
+    jq --arg v "$value" ".${field} = \$v" "$APP_PROPS_FILE" >tmp.json && mv tmp.json "$APP_PROPS_FILE"
+  fi
+}
+
+# Update fields if corresponding environment variables are set
```

Comment:
can we also automate this reading the `appProps.template.json` and mapping the env variable? when adding new attributes we then have con change:
- appProps.template.json
- Readme.md
- Dockerfile
- entrypoint.sh